### PR TITLE
Better document step-22 solvers.

### PR DESCRIPTION
**Pull request by bangerth**
I've found the need to point out too many times that the solver used in step-22 is not the way to go, and that instead one should use the one discussed in the "Possibilities for extension" section. Make this clear in the introduction.

/rebuild

### CODE REVIEW

**Comment by masterleinad** [comment]
```
/home/runner/work/dealii/dealii/build/doc/doxygen/tutorial/step-22.h:685: warning: unable to resolve reference to `SW94' for \ref command
/home/runner/work/dealii/dealii/build/doc/doxygen/tutorial/step-22.h:686: warning: unable to resolve reference to `elman2005' for \ref command
```

File: examples/step-22/doc/intro.dox [hunk 1]
```diff
@@ -588,6 +588,25 @@ classes, we will be able to use them interchangeably using the same syntax in
...
+way this system of equations *should* be solved?
+
+The answer to this is no. The primary bottleneck with the approach,
+already identified above, is that we have to repeatedly solve linear
+systems with $A$ inside the Schur complement, and because we don't
+have a good preconditioner for the Schur complement, these solves just
+have to happen too often. A better approach is to use a block
+decomposition, which is based on an observation of Silvester and
+Wathen @ref SW94 and explained in much greater detail in
+@ref elman2005 . An implementation of this alternative approach is
```
Comment (masterleinad) [inline examples/step-22/doc/intro.dox:605]:
```suggestion
@cite elman2005 . An implementation of this alternative approach is
```

File: examples/step-22/doc/intro.dox [hunk 2]
```diff
@@ -588,6 +588,25 @@ classes, we will be able to use them interchangeably using the same syntax in
...
+tutorial programs are teaching tools that makes sense. But is this the
+way this system of equations *should* be solved?
+
+The answer to this is no. The primary bottleneck with the approach,
+already identified above, is that we have to repeatedly solve linear
+systems with $A$ inside the Schur complement, and because we don't
+have a good preconditioner for the Schur complement, these solves just
+have to happen too often. A better approach is to use a block
+decomposition, which is based on an observation of Silvester and
+Wathen @ref SW94 and explained in much greater detail in
```
Comment (masterleinad) [inline examples/step-22/doc/intro.dox:604]:
```suggestion
Wathen @cite SW94 and explained in much greater detail in
```

**Review by masterleinad — COMMENTED** [review]
Otherwise, this looks good to me.

**Comment by bangerth** [comment]
Ah, darn. Thanks for pointing it out -- fixed now!

**Review by masterleinad — APPROVED** [review]

**Comment by bangerth** [comment]
The windows build has timed out, but the rest of it seems to all succeed.

**Pull request merged**